### PR TITLE
Rebalances the PPSh and readds it to normal vendors.

### PR DESCRIPTION
**Pull request by SplinterGP**
## About The Pull Request
Rebalances the PPSh and its magazines, now the PPSh has a bit better stats overall with the normal mags.

Normal mags now have the size small, like the T-90 does and had a bit of an ammo increase.
The drum mags now have the size NORMAL like T-12 mags and had a bit of ammo increase as well, it now also increases the gun aim slowdown, scatter, unwielded scatter and wield delay, trading some efficiency for more ammo.
and the PPSh now has more attachments to use.

This also removes PPSh from seasonals and puts it back on normal vendors, since its the only sidegrade there is to the T-90 overall, the T-19 being more of a sidearm than an actual SMG.

## Why It's Good For The Game
Rebalances the PPSh and gives it some unique quirk of extended mags stat + some overall usability when compared to the T-90/any other equivalent gun, it is now overall a hybrid of a SMG/Carbine, being able to use mags with more ammo but being slower overall when doing so.

## Changelog
:cl:
balance: PPSh can now take way more attachments than before.
balance: PPSh stats change to take in account the magazine stats + overall buff to make it competitive with the T-90
balance: PPSh extended mags now provide scatter increase and, wield and movement speed debuffs
balance: PPSh magazines now have proper size, the normal mags being SMALL like the T-90 magazines and extended mags being NORMAL like T-12 magazines.
/:cl:

### CODE REVIEW

**Comment by Pariah919** [comment]
ppsh fits more in seasonals than it does inside normal marine vendors mechanically and visually considering it has next to no attachment options, doesn't really consitute it's own niche and feels more like a half done gimmick
also you didn't replace anything on the mkh seasonal which breaks the pattern of 2 guns

**Comment by BraveMole** [comment]
MJP just did a PPSh rebalance. For the putting back into vendor, ask david first

**Comment by SplinterGP** [comment]
> 
> 
> ppsh fits more in seasonals than it does inside normal marine vendors mechanically and visually considering it has next to no attachment options, doesn't really consitute it's own niche and feels more like a half done gimmick
> also you didn't replace anything on the mkh seasonal which breaks the pattern of 2 guns

i just gave it attachments doe.

**Comment by Sirsnek15** [comment]
> ppsh fits more in seasonals than it does inside normal marine vendors mechanically and visually considering it has next to no attachment options, doesn't really consitute it's own niche and feels more like a half done gimmick
> also you didn't replace anything on the mkh seasonal which breaks the pattern of 2 guns

There's literally no reason for it to be in seasonals, and it shouldn't have been given up just to fit a "pattern" in the first place.

File: code/modules/projectiles/ammunition.dm [hunk 1]
```diff
@@ -85,6 +96,19 @@ They're all essentially identical when it comes to getting the job done.
 
 		transfer_ammo(H, user, H.current_rounds) // This takes care of the rest.
 
+/obj/item/ammo_magazine/proc/on_inserted(obj/item/weapon/gun/master_gun)
```
Comment (BraveMole) [inline code/modules/projectiles/ammunition.dm:99]: autodoc

File: code/modules/projectiles/ammunition.dm [hunk 2]
```diff
@@ -85,6 +96,19 @@ They're all essentially identical when it comes to getting the job done.
...
+	master_gun.scatter						+= scatter_mod
+	master_gun.scatter_unwielded			+= scatter_unwielded_mod
+	master_gun.aim_slowdown					+= aim_speed_mod
+	master_gun.wield_delay					+= wield_delay_mod
+
+
+/obj/item/ammo_magazine/proc/on_removed(obj/item/weapon/gun/master_gun)
```
Comment (BraveMole) [inline code/modules/projectiles/ammunition.dm:106]: autodoc

**Review by BraveMole — CHANGES_REQUESTED** [review]

**Pull request closed without merging**